client/signup/steps/difm-site-picker/index.tsx

```diff
@@ -0,0 +1,196 @@
+		return !! (
+			site.capabilities?.manage_options &&
+			! site.jetpack &&
+			! site.options?.is_wpforteams_site &&
+			! site.options?.is_difm_lite_in_progress
+		);
+	};
+
+	const onCloseDialog = () => {
+		setSiteId( null );
+	};
+
+	const onConfirmDelete = () => {
+		dispatch(
+			submitSignupStep(
+				{
+					stepName: props.stepName,
+					stepSectionName: props.stepSectionName,
+					siteId,
+					siteSlug,
+				},
+				{
+					siteId,
+					siteSlug,
+				}
+			)
+		);
+		//Skip domains step
+		dispatch(
+			submitSignupStep(
+				{
+					stepName: 'domains',
+					wasSkipped: true,
+				},
+				{
+					domainItem: undefined,
+					themeItem: undefined,
+				}
+			)
+		);
+
+		goToNextStep();
+	};
+
+	if ( siteId ) {
+		const deleteDisabled =
+			typeof confirmDomain !== 'string' ||
+			confirmDomain.toLowerCase().replace( /\s/g, '' ) !== siteDomain;
+		const buttons = [
+			<Button onClick={ onCloseDialog }>{ translate( 'Cancel' ) }</Button>,
+			<Button primary scary disabled={ deleteDisabled } onClick={ onConfirmDelete }>
+				{ translate( 'Delete site content' ) }
+			</Button>,
+		];
+
+		return (
+			<Dialog isVisible={ true } buttons={ buttons } onClose={ onCloseDialog }>
+				<h1>{ translate( 'Site Reset Confirmation' ) }</h1>
+				<p>{ translate( 'After completing your purchase:' ) }</p>
+				<ul>
+					<li>
+						{ translate(
+							'The current content of your website {{strong}}%(siteTitle)s{{/strong}} (%(siteAddress)s) will be deleted, so we can start your website build on an empty site. ' +
+								'This includes pages, posts, media, and comments.',
+							{
+								components: {
+									strong: <strong />,
+								},
+								args: {
+									siteTitle,
+									siteAddress: siteDomain,
+								},
+							}
+						) }
+					</li>
+					<li>
+						{ translate(
+							'Any WordPress.com subscriptions on your site such as custom domains and emails will not be affected.'
+						) }
+					</li>
+					<li>
+						{ translate(
+							'Your site will be inaccessible in {{i}}Coming Soon{{/i}} mode while we build your new site.',
+							{
+								components: {
+									i: <i />,
+								},
+							}
+						) }
+					</li>
+				</ul>
+				<p>
+					{ translate(
+						'If you don’t want your site content to be deleted, we can instead create a new site on your account for the website build. ' +
+							'You can then transfer the existing content to the new site once it’s complete. ' +
+							'Click {{a}}here{{/a}} to go back and choose "New Site".',
+						{
+							components: {
+								a: <a href="/start/do-it-for-me" />,
+							},
+						}
+					) }
+				</p>
+				<FormLabel htmlFor="confirmDomainChangeInput">
+					{ translate(
+						"Please type {{warn}}%(siteAddress)s{{/warn}} to acknowledge that after purchase all of your site's content will be permanently deleted.",
+						{
+							components: {
+								warn: <span />,
+							},
+							args: {
+								siteAddress: siteId && siteDomain,
```

Comment:
All this logic is wrapped in a `if( siteId )` block, so the `sideId &&` check here is unnecessary and can be removed.